### ISSUE TITLE
Implement better LazyElement logging

### ISSUE THREAD

**Issue by FermJacob** [Magenic/MAQS#39]
I noticed that we don't include the action name (e.g. Click(), GetValue(), Submit()) in the LazyElement error logs. It appears that info can be easily made available in most/all situations. In other frameworks I've written, users have found this added info to be valuable.
 
The "top-level" exception's message could look something like:
ProfileLink could not be found when you called Click

<pre><code> private IWebElement GetElement(Func<IWebElement> getElement, [CallerMemberName] string callerName = "")
        {
            // Try to use cached element
            if (this.CachedElement != null)
            {
                try
                {
#pragma warning disable S1481
                    // Do this to make sure we waited for the element to exist
                    bool visible = this.CachedElement.Displayed;
#pragma warning restore S1481
                    return this.CachedElement;
                }
                catch (Exception e)
                {
                    this.TestObject.Log.LogMessage(MessageType.VERBOSE, "Refinding element because: " + e.Message);
                }
            }
            try
            {
                this.TestObject.Log.LogMessage(MessageType.VERBOSE, "Performing lazy driver find on: " + this.By);
                this.CachedElement = getElement();
                return this.CachedElement;
            }
            catch (Exception e)
            {
                StringBuilder messageBuilder = new StringBuilder();
 
                messageBuilder.AppendLine(this.userFriendlyName + " could not be found when you called " + callerName);
 
                messageBuilder.AppendLine("Failed to find: " + this.userFriendlyName);
                messageBuilder.AppendLine("Locator: " + this.By);
                messageBuilder.AppendLine("Because: " + e.Message);
 
                throw new Exception(messageBuilder.ToString(), e);
            }
        }</code></pre>

**Comment by FermJacob** [comment]
@TroyWalshProf Looks like 

`private void ExecuteEvent(Action elementAction, string caller)`

Takes in a caller to log, but GetElement doesn't.  Do we want to use CallerMemberName or pass in the caller?

**Comment by magenicgrizzaffi** [comment]
Hmmm...looking at the call sites for ExecuteEvent, I see what's going on there. Could we use CallerMemberName in ExecuteEvent? Will it work appropriately with those lambda expressions?
![image](https://user-images.githubusercontent.com/55812056/65637467-d7881e80-dfa9-11e9-9a58-3c075509def8.png)

**Comment by TroyWalshProf** [comment]
I don't think we want to change the logging for this.  

`private IWebElement GetElement(Func<IWebElement> getElement)`
Basically just finds the element. 
So this is before we try to interact with it.

To put it another way
If the an element doesn't exist why would you want to include information on what you were planning to do with it after you found it?

**Comment by magenicgrizzaffi** [comment]
It adds extra context. It may lower the bar for entry for those who are tasked with triaging automation failures, but who may not be code savvy.

I’m not going to go to the mat for this; I don’t think it’ll add like 50% productivity to most of our users. I’ve had previous users who found it valuable, but our users are not those users so BYOS. 😉

**Comment by TroyWalshProf** [comment]
I can make better logging that looks like this:
```
2019-10-03 14:51:01.928
ERROR:	
Timed out waiting for <Click>b__34_0 to return
   at Magenic.Maqs.Utilities.Helper.GenericWait.Wait[T](Func`1 waitFor, TimeSpan retryTime, TimeSpan timeout) in C:\Users\TroyW\source\repos\Magenic\MAQS\Framework\Utilities\Helper\GenericWait.cs:line 393
   at Magenic.Maqs.BaseSeleniumTest.Extensions.LazyElement.Click() in C:\Users\TroyW\source\repos\Magenic\MAQS\Framework\BaseSeleniumTest\Extensions\LazyElement.cs:line 212
   at SeleniumUnitTests.LazyElementUnitTests.TESTME2() in C:\Users\TroyW\source\repos\Magenic\MAQS\Framework\SeleniumUnitTests\LazyElementUnitTests.cs:line 177
Failed to find: Disabled
Locator: By.CssSelector: #disabledField INPUT
Because: Timed out after 2 seconds: Waiting for clickable element
```

Would that address the want?

**Comment by billros** [comment]
Is “Disabled” the friendly name from the LayElement?

I’m with Paul, if we can make the logs provide Friendly messages (I don’t consider the “By” value friendly) it looks cleaner.

**Comment by magenicgrizzaffi** [comment]
Follow up.
Troy and I had a good chat about this. We decided that we should “What we want for logging and reporting?” instead of “how do we add to what we already have for reporting?” We have some ideas about how to approach that. More to come soon.

@Troy Walsh<mailto:TroyW@magenic.com>, should Jake close the current work item or keep it open for this kind of convo?

Thanks everyone!
Paul

**Comment by FermJacob** [comment]
This will be deployed with MAQS 5.6

**Comment by TroyWalshProf** [comment]
My "fix" is a small band-aid.
We need a higher level whiteboarding session when Paul get's into town later this month

**Comment by TroyWalshProf** [comment]
@magenicgrizzaffi and @billros 
I think this was already addressed with #60  and #162 (Adding Steps and Actions)
Please let me know if you both see it differently
